File: CSharpStuff/Nathandelane.Math.Processor/Nathandelane.Math.Processor.Tokens/AllTokens.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Nathandelane.Math.Processor.Tokens
{
	public class AllTokens
	{
		#region Fields

		private static IList<IToken> _tokens = new List<IToken>()
		{
			new Number(0),
			new OpenPerenthesis(),
			new ClosePerenthesis(),
			new Multiplication(),
			new Division(),
			new Subtraction(),
			new Addition(),
			new Sine(),
			new Cosine(),
			new Tangent()
		};

		#endregion

		#region Properties

		public static IList<IToken> Set
		{
			get { return _tokens; }
		}

		#endregion
	}
}

File: JavaStuff/PersonalCalculator/personalcalculator/src/com/nathandelane/math/evaluation/tokens/AllTokens.java
package com.nathandelane.math.evaluation.tokens;

import java.util.ArrayList;
import java.util.Collections;

public class AllTokens {

	private static ArrayList<IToken> __tokens = null;
	
	public static ArrayList<IToken> getSet() {
		if(__tokens == null) {
			__tokens = new ArrayList<IToken>();
			
			Collections.addAll(__tokens,
				new SetCommand(),
				new Assignment(),
				new Number(0),
				new Pi(),
				new E(),
				new LastResult(),
				new OpenPerenthesis(),
				new ClosePerenthesis(),
				new Multiplication(),
				new Division(),
				new Subtraction(),
				new Addition(),
				new Sine(),
				new Cosine(),
				new Tangent()
			);
		}
		
		return __tokens;
	}
	
}
